packages/ts/react-crud/src/autoform-field.tsx

```diff
@@ -1,36 +1,103 @@
+import { type EnumModel, _enum } from '@hilla/form';
+import { Checkbox, type CheckboxProps } from '@hilla/react-components/Checkbox.js';
+import { DatePicker, type DatePickerProps } from '@hilla/react-components/DatePicker.js';
+import { DateTimePicker, type DateTimePickerProps } from '@hilla/react-components/DateTimePicker.js';
+import { IntegerField, type IntegerFieldProps } from '@hilla/react-components/IntegerField.js';
+import { NumberField, type NumberFieldProps } from '@hilla/react-components/NumberField.js';
+import { Select, type SelectProps } from '@hilla/react-components/Select.js';
+import { TextField, type TextFieldProps } from '@hilla/react-components/TextField.js';
+import { TimePicker, type TimePickerProps } from '@hilla/react-components/TimePicker.js';
 import type { UseFormResult } from '@hilla/react-form';
 import type { JSX } from 'react';
 import type { PropertyInfo } from './property-info.js';
 
-interface AutoFormFieldProps {
+export type SharedFieldProps = Readonly<{
   propertyInfo: PropertyInfo;
   form: UseFormResult<any>;
-  disabled?: boolean;
-}
+}>;
 
 function getPropertyModel(form: UseFormResult<any>, propertyInfo: PropertyInfo) {
   // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
   return form.model[propertyInfo.name];
 }
 
-function AutoFormTextField({ propertyInfo, form, disabled }: AutoFormFieldProps) {
+type AutoFormTextFieldProps = SharedFieldProps & TextFieldProps;
+
+function AutoFormTextField({ propertyInfo, form, ...other }: AutoFormTextFieldProps) {
+  const model = getPropertyModel(form, propertyInfo);
+  // @ts-expect-error: TODO: fix the typing issue
+  return <TextField {...other} {...form.field(model)} label={propertyInfo.humanReadableName} />;
+}
+
+type AutoFormIntegerFieldProps = IntegerFieldProps & SharedFieldProps;
+
+function AutoFormIntegerField({ propertyInfo, form, ...other }: AutoFormIntegerFieldProps) {
+  const model = getPropertyModel(form, propertyInfo);
+  // @ts-expect-error: TODO: fix the dangerouslySetInnerHTML typing issue
+  return <IntegerField {...other} {...form.field(model)} label={propertyInfo.humanReadableName} />;
+}
+
+type AutoFormNumberFieldProps = NumberFieldProps & SharedFieldProps;
+
+function AutoFormDecimalField({ propertyInfo, form, ...other }: AutoFormNumberFieldProps) {
+  const model = getPropertyModel(form, propertyInfo);
+  // @ts-expect-error: TODO: fix the dangerouslySetInnerHTML typing issue
+  return <NumberField {...other} {...form.field(model)} label={propertyInfo.humanReadableName} />;
+}
+
+type AutoFormDateFieldProps = DatePickerProps & SharedFieldProps;
+
+function AutoFormDateField({ propertyInfo, form, ...other }: AutoFormDateFieldProps) {
+  const model = getPropertyModel(form, propertyInfo);
+  // @ts-expect-error: TODO: fix the dangerouslySetInnerHTML typing issue
+  return <DatePicker {...other} {...form.field(model)} label={propertyInfo.humanReadableName} />;
```

Comment:
Date pickers should use the same date format as the one in the header filter.

![Bildschirmfoto 2023-10-26 um 12 21 39](https://github.com/vaadin/hilla/assets/357820/fcf16858-161f-406e-830e-bd1dbc1444c8)

Reply:
Should be OK now